src/test/java/com/gradle/enterprise/fixtures/DevelocityConfigurationForTest.java

```diff
@@ -0,0 +1,89 @@
+package com.gradle.enterprise.fixtures;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.gradle.develocity.agent.gradle.DevelocityConfiguration;
+import com.gradle.develocity.agent.gradle.buildcache.DevelocityBuildCache;
+import com.gradle.develocity.agent.gradle.scan.BuildScanConfiguration;
+import org.gradle.api.Action;
+import org.gradle.api.provider.Property;
+
+import javax.annotation.Nullable;
+
+public class DevelocityConfigurationForTest implements DevelocityConfiguration {
```

Comment:
🤔 Can't we use a regular mocking framework like Mockito or Spock?

Reply:
Is there a better way to test this? And even if we use a fake develocity plugin, why can't the fake develocity plugin use real `Property` instances instead of this strange `ProxyProperty`?